pkg/network/encoding/marshal/usm_encoders_linux.go

```diff
@@ -0,0 +1,34 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025-present Datadog, Inc.
+
+//go:build linux && linux_bpf
+
+package marshal
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/network"
+)
+
+func initializeUSMEncoders(conns *network.Connections) []usmEncoder {
+	encoders := make([]usmEncoder, 0)
+
+	if encoder := newHTTPEncoder(conns.HTTP); encoder != nil {
```

Comment:
Should it be configuration-aware? In other words, should we only initialize an encoder if the corresponding protocol is enabled in the configuration?

Reply:
One possible optimization is to filter the list upfront to include only the enabled protocols. However, the performance gain seems minimal, so we can defer it for now

Reply:
I agree with the later, as all encoders check the payload they are getting are not empty. So the check is minimal.
Since we're talking on a very limited number of encoders, the preconfigured list seems like a very small gain.